encoders/firebase-encoders-reflective/src/main/java/com/google/firebase/encoders/reflective/ReflectiveDecoderHelper.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.encoders.reflective;
+
+import com.google.firebase.encoders.EncodingException;
+import com.google.firebase.encoders.FieldDescriptor;
+import com.google.firebase.encoders.annotations.Encodable;
+import com.google.firebase.encoders.annotations.ExtraProperty;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+class ReflectiveDecoderHelper {
+  private ReflectiveDecoderHelper() {}
+
+  static FieldDescriptor buildFieldDescriptor(String decodingKey, Annotation[] annotations) {
+    FieldDescriptor.Builder builder = FieldDescriptor.builder(decodingKey);
+    for (Annotation annotation : annotations) {
+      if (annotation.annotationType().isAnnotationPresent(ExtraProperty.class)) {
+        builder.withProperty(annotation);
+      }
+    }
+    return builder.build();
+  }
+
+  static boolean isInLine(FieldDescriptor fieldDescriptor) {
+    Encodable.Field annotation = fieldDescriptor.getProperty(Encodable.Field.class);
+    return annotation != null && annotation.inline();
+  }
+
+  static String fieldName(Method method) {
```

Comment:
We want to make sure the field name is consistent with whatever the reflective *encoder* does, can you make sure they are indeed consistent and if not, reconsile them?

Reply:
yes, they are consistent, but i will change the implementation to align with encoder as well.